packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -747,8 +747,20 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
 
   // Calculate the new scroll offset so the cursor remains visible.
   double _getScrollOffsetForCaret(Rect caretRect) {
-    final double caretStart = _isMultiline ? caretRect.top : caretRect.left;
-    final double caretEnd = _isMultiline ? caretRect.bottom : caretRect.right;
+    double caretStart;
+    double caretEnd;
+    if (_isMultiline) {
+      caretStart = caretRect.top;
+      // The caret is vertically centered in the line with an offset. Include this
```

Comment:
The caret is vertically centered within the line. Expand the caret's height so that it spans the line because we're going to ensure that the entire expanded caret is scrolled into view.